main.go

```diff
@@ -253,13 +253,10 @@ func visit(url *url.URL) {
 	client := &http.Client{
 		Transport: tr,
 		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			if !followRedirects {
-				return http.ErrUseLastResponse
-			}
-			if len(via) >= maxRedirects {
+			if redirectsFollowed >= maxRedirects {
```

Comment:
this logic can be simplified to 
```go
return http.ErrUseLastResponse
```
as the redirectsFollowed check is part of visit().